lib/CodeBuilder/Adapter/TolerantParser/Updater/AbstractMethodUpdater.php

```diff
@@ -91,6 +95,19 @@ public function updateMethods(Edits $edits, ClassLikePrototype $classPrototype,
         }
 
         foreach ($methodPrototypes as $methodPrototype) {
+            // If class has methods add the constructor before it.
+            if ($methodPrototype->name() === '__construct' && $firstMethod !== null) {
+                // We can't use the first method's starting location because this is already indented to we unindent the code
+                $edits->add(
+                    TextEdit::create(
+                        $firstMethod->getStartPosition() - 4,
```

Comment:
Maybe there is a better text token I could use?

Reply:
the indentation is configurable, so we can't depend on -4. the current code always starts _after_ and adds a new line. If we could determine the last non-method node then we can do it the same way.